### PR TITLE
refactor(compiler): don’t write summaries for jit by default

### PR DESCRIPTION
**Pull request by tbosch**
The default is false externally but true internally at Google.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by vicb — APPROVED** [review]

File: packages/compiler-cli/src/codegen.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export class CodeGenerator {
       i18nFormat: cliOptions.i18nFormat,
       locale: cliOptions.locale, missingTranslation,
       enableLegacyTemplate: options.enableLegacyTemplate !== false,
+      enableSummariesForJit: options.enableSummariesForJit !== false,
```
Comment (IgorMinar) [inline packages/compiler-cli/src/codegen.ts]: why is this option JIT specific?

My understanding of the original issue was that generating NgSummaries by default even in AOT mode is problematic. Did I misunderstand the problem?

File: packages/compiler-cli/test/main_spec.ts [hunk 2]
```diff
@@ -261,5 +261,52 @@ describe('compiler-cli', () => {
...
+          .then((exitCode) => {
+            expect(exitCode).toEqual(0);
+            expect(fs.existsSync(path.resolve(outDir, 'mymodule.ngsummary.js'))).toBe(false);
+
+            done();
+          })
+          .catch(e => done.fail(e));
+    });
+
+    it('should not produce ngsummary files if configured', (done) => {
```
Comment (alxhub) [inline packages/compiler-cli/test/main_spec.ts]: The test name is incorrect - it is asserting that it *should* produce ngsummary files if configured.

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/src/codegen.ts (hunk 1 above)
Comment (tbosch) [inline packages/compiler-cli/src/codegen.ts]: We have .ngsummary.json files, which are only for AOT.
But we added `.ngsummary.ts` file that allow to use AOTed code in JIT (more specific in tests)

Comment (IgorMinar) [inline packages/compiler-cli/src/codegen.ts]: I see. The option name doesn't reflect that well. how about `generateSummarySources` ?

File: packages/compiler-cli/test/main_spec.ts (hunk 2 above)
Comment (tbosch) [inline packages/compiler-cli/test/main_spec.ts]: Done

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**